libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -510,3 +630,246 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
 		);
 
 }
+
+void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _contract) const
+{
+	vector<FunctionDefinition const*> contractFuncs = _contract.definedFunctions();
+
+	auto const resolvedBases = resolveBaseContracts(_contract);
+
+	FunctionSet collectedFunctions;
+
+	// Collect the most inherited functions of the direct base contracts
+	for (ContractDefinition const* contract: resolveBaseContracts(_contract))
+	{
+		FunctionSet baseSet = convertContainer<FunctionSet>(contract->definedFunctions());
+
+		for (FunctionDefinition const* function: getBaseFunctions(contract))
+		{
+			if (baseSet.cend() == baseSet.find(function))
+				baseSet.insert(function);
+		}
+
+		if (baseSet.empty())
+			continue;
+
+		// Collect the first of every override-set of functions (they are sorted
+		// by most-inherited function)
```

Comment:
This is kind of implicit in all the functions - is it really correct and will it stay like that?

Reply:
upon closer inspections, the whole loop creating the `collectedFunctions` seems to be redundant now and we can just use `getBaseFunctions()` directly.